### PR TITLE
feat(accessibility): fix roles and console's warnings

### PR DESCRIPTION
**Pull request by shmodarresi**

### CODE REVIEW

File: src/pages/MainPage/MainPage.tsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ export const MainPage: React.FC<MainPageProps> = ({ title, children, description
           profileLinks={profileLinks}
         />
       </CustomHeader>
-      <main>
+      <main style={{ marginBottom: '2.5rem' }}>
```
Comment (shekhar-shubhendu) [inline src/pages/MainPage/MainPage.tsx]: Please change this to styled component

File: src/design-system/molecules/Footer/Footer.tsx [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
+  margin-top: auto;
   position: relative;
-  top: auto;
-  bottom: 0;
   padding: 2.5rem;
+  box-sizing: border-box;
+  background-color: ${({ theme }) => theme.palette.background.default};
 `;
 
+const FooterContainer = styled(Container)``;
```
Comment (shekhar-shubhendu) [inline src/design-system/molecules/Footer/Footer.tsx]: why do we want to do this?
maybe just use `Container` directly.

**Comment by tlebon** [comment]
tests failing also

**Review by shekhar-shubhendu — DISMISSED** [review]

**Review by shekhar-shubhendu — APPROVED** [review]

File: src/design-system/atoms/Button/Button.tsx [hunk 3]
```diff
@@ -83,8 +83,8 @@ export const CustomButton: React.FC<ButtonProps> = ({
       startIcon={iconPosition === 'left' ? icon : null}
       endIcon={iconPosition === 'right' ? icon : null}
       bordercolor={internalBorderColor}
-      lowercase={lowercase}
-      sx={customStyle}
+      texttype={lowercase ? 'none' : 'uppercase'}
+      sx={{ ...customStyle }}
```
Comment (cpooley) [inline src/design-system/atoms/Button/Button.tsx:87]: Why did sx={customStyle} get switched to sx={{...customStyle}}?

Comment (shmodarresi) [reply src/design-system/atoms/Button/Button.tsx:87]: to assign an object

**Pull request merged**